README.md

```diff
@@ -1,2 +1,203 @@
+
+- Creating module manifest file :
+```
+module:
+  name: 'modulename'
+  packageurl: 'link/to/github/repository'
+  tag: 0.0.0 
+  path: 'path/to/module/in/system'
+  publicheaders: 'if present'
+  sources: 'if present'
+  targets: 'modulename'
+  deps: 'dependencies'
+```
+
+- Example module.yml :
+```
+module: 
+  name: FFTW
+  packageurl: "https://github.com/root-project/root/tree/master/math/fftw"
+  tag: 0.0.0 
+  path: /home/username/root/math/fftw
+  publicheaders: 
+  sources: no_sources
+  targets: FFTW
+  deps: Core
+```  
+
+- Creating package manifest file :
+```
+package:
+  name: 
+  tag: 'version value'
+  targets:
+    target:
+    name: 'packagename'
+  products:
+    package:
+      name: 
+  module1:
+    name: 'modulename'
+    packageurl: 'link/to/github/repository'
+    tag: 0.0.0 
+    path: 'path/to/module/in/system'
+    publicheaders: 'if present'
+    sources: 'if present'
+    targets: 'modulename'
+    deps: 'dependencies'
+  module2:
+    .
+    .
+  module:
+```
+
+- Example packagename.yml :
+```
+package: 
+ name: "IO"
+ tag: 0.0.0
+ targets:
+  target:
+   name: "IO"
+ products:
+  package:
+   name: IO
+ module: 
+  name: "RCastor"
+  packageurl: "https://github.com/root-project/io/castor"
+  tag: 0.0.0
+  path: /home/ashwin/root/io/castor
+  publicheaders: 
+  sources: no_sources
+  targets: 
+  deps: Net;Core;RIO
+ module: 
+  name: "DCache"
+  packageurl: "https://github.com/root-project/io/dcache"
+  tag: 0.0.0
+  path: /home/ashwin/root/io/dcache
+  publicheaders: 
+  sources: no_sources
+  targets: 
+  deps: Net;Core;RIO 
+```
+
+## Prerequisites
+
+python3
+
+[matplotlib](https://matplotlib.org/users/installing.html)
+
+[networkx](https://networkx.github.io/documentation/stable/install.html)
+
+pyyaml (install with - sudo apt-get install python-yaml)
+
+cmake (Version 3.8 & above)
+
+
+## Initial setup
+Required - root build files directory, if not present already, do an out of source build for root
+
+(if root is present in '/home/username/root' then inside '/home/username' do the following)
+```
+$ mkdir rootbuild
+$ cd rootbuild
+$ cmake -DCMAKE_CXX_STANDARD=14 -Dbuiltin_clang=on ../root
+$ make
+```
+Fork and Clone the repository (preferably in '/home/username' directory)
+```
+$ git clone https://github.com/oshadura/root-get
+```
+Create a directory where you would like to install new packages using root-get
+(inside '/home/username' directory or any other path)
+```
+$ mkdir installdir
+```
+Inside the root-get directory, set environment variables
+```
+$ cd root-get
+$ export ROOTSOURCES="/path/to/root/source/directory"
+$ export ROOT_PKG_PATH="/path/where/to/install/new/packages"
+(/home/username/installdir if you have followed above method)
+$ export ROOTSYS="/path/to/rootbuild/directory"
+$ export ROOTSYS="/path/to/module/manifest/files"
+```
+
+## Installing
+
+To install a new module with root-get :
```

Comment:
Can we add here all other options we have? "-mi", "-pi", "-s", "-l" and etc.

Reply:
Yes, I'll do it. with subheadings for each of these.